frontend/src/pages/BookDetails.tsx

```diff
@@ -49,15 +53,45 @@ export default function BookDetails(props: BookDetailsProps) {
             .catch(console.error)
     }
 
+    function handleCheckoutBook() {
+
+        props.checkoutBook(user?.id || "undefined", id || "undefined")
+            .then(() => {
+                toast.success('📖 Book is checkout.Enjoy!', {
+                    position: "top-right",
+                    autoClose: 5000,
+                    hideProgressBar: false,
+                    closeOnClick: true,
+                    pauseOnHover: true,
+                    draggable: true,
+                    progress: undefined,
+                    theme: "light",
+                });
+            })
+            .catch((error) => {
+                toast.error('📚 Sorry.You can not loan the Book!', {
+                    position: "top-right",
+                    autoClose: 5000,
+                    hideProgressBar: false,
+                    closeOnClick: true,
+                    pauseOnHover: true,
+                    draggable: true,
+                    progress: undefined,
+                    theme: "light",
+                });
+                console.error(error)
+            })
+    }
+
     return (
         <Layout>
             <h1 className={"h1-book-gallery"}>Book <span>Details</span></h1>
-        <div className={"book-card"}>
-            <div className={"book-card-div-two"}>
-                {book.img ? <img src={book.img} alt="Book"/> : <img src={"/book.jpg"} alt="Book"/>}
+            <div className={"book-card"}>
+                <div className={"book-card-div-two"}>
+                    {book.img ? <img src={book.img} alt="Book"/> : <img src={"/book.jpg"} alt="Book"/>}
```

Comment:
maybe you can use something like that to have a picture in case of error or missing :` const [imgSrc, setImgSrc] = useState(props.pet.photo);
    return (
        <div className={"Card"}>
      
            <img src={imgSrc} onError={()=> setImgSrc("/petfit_logo_small_icon_only_inverted.png")} alt={"logo"}/><br/>
       
        </div>
    )`